### PR TITLE
ContactGroupAssign DataService

### PR DESCRIPTION
**Pull request by freezey**

### CODE REVIEW

File: Infusionsoft/ContactGroupAssign.php [hunk 1]
```diff
@@ -1,7 +1,24 @@
-<?php
-class Infusionsoft_ContactGroupAssign extends Infusionsoft_Generated_ContactGroupAssign{	
+<?php
+class Infusionsoft_ContactGroupAssign extends Infusionsoft_Generated_ContactGroupAssign{
+
     public function __construct($id = null, $app = null){
```
Comment (freezey) [inline Infusionsoft/ContactGroupAssign.php]: I recommend a comment block explaining how $id works

File: Infusionsoft/ContactGroupAssign.php [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+        if($id != null) {
+            $results = Infusionsoft_ContactGroupAssignDataService::query(new Infusionsoft_ContactGroupAssign(), array('GroupId' => $id % 10000000, 'ContactId' => floor($id / 10000000)), 1, 0, array('ContactId', 'GroupId', 'DateCreated', 'ContactGroup'), $app);
+            if (count($results) == 0) {
+                throw new Infusionsoft_Exception("Could not load " . $this->table . " with id " . $id);
+            }
+            $object = $results[0];
+
+
+
+            $this->GroupId = $object->GroupId;
```
Comment (freezey) [inline Infusionsoft/ContactGroupAssign.php]: We should be able to use ::load() for this. If we can't, can you add a comment to explain?

File: Infusionsoft/ContactGroupAssignDataService.php [hunk 3]
```diff
@@ -1,26 +1,37 @@
...
+
         $results = Infusionsoft_DataService::queryWithOrderBy($object, $queryData, $orderByField, $ascending, $limit, $page, $returnFields, $app);
+
         Infusionsoft_ContactGroupAssign::addCustomField('Id');
         self::addCompoundKeyToResults($results);
 
-        if($orderByField == 'ContactId' && $ascending){
+        if($orderByField == 'ContactId' && $ascending && count($results) > 0){
             if($page > 0){
                 self::removeRecordsForContactsAlreadyProcseed(self::$lastProcessedContactId, $results);
```
Comment (freezey) [inline Infusionsoft/ContactGroupAssignDataService.php]: typo in method call

File: Infusionsoft/ContactGroupAssignDataService.php [hunk 4]
```diff
@@ -56,7 +70,7 @@ public static function addCompoundKeyToResults(array &$results){
     public static function removeRecordsForContactsAlreadyProcseed($lastProcessedContactId, &$results){
```
Comment (freezey) [inline Infusionsoft/ContactGroupAssignDataService.php]: typo in method name

File: Infusionsoft/ContactGroupAssign.php (hunk 1 above)
Comment (joeynovak) [inline Infusionsoft/ContactGroupAssign.php]: Good point.  I'll think about it.

File: Infusionsoft/ContactGroupAssign.php (hunk 2 above)
Comment (joeynovak) [inline Infusionsoft/ContactGroupAssign.php]: I need to think about this some more.  The reason is that load only works with tables that have an Id column...  And this table doesn't.  I'm not 100% we should comment everything that we do differently...  But, idk yet.  I need to think about it.

**Pull request merged**